File: capture-nexus-screenshots.py
#!/usr/bin/env python3
import asyncio
import os
from pathlib import Path
from playwright.async_api import async_playwright

async def capture_screenshots():
    screenshots_dir = Path('./screenshots')
    screenshots_dir.mkdir(exist_ok=True)
    
    pages = [
        ('http://127.0.0.1:9696/nexus/', 'nexus-hub'),
        ('http://127.0.0.1:9696/nexus/nexus-genesis.html', 'nexus-genesis'),
        ('http://127.0.0.1:9696/nexus/nexus-autonomy.html', 'nexus-autonomy'),
        ('http://127.0.0.1:9696/nexus/nexus-discovery.html', 'nexus-discovery'),
        ('http://127.0.0.1:9696/nexus/nexus-integration.html', 'nexus-integration'),
    ]
    
    async with async_playwright() as p:
        browser = await p.chromium.launch()
        
        # Desktop 1920x1080
        desktop_context = await browser.new_context(viewport={'width': 1920, 'height': 1080})
        for url, name in pages:
            page = await desktop_context.new_page()
            await page.goto(url, wait_until='networkidle')
            await page.screenshot(path=f'{screenshots_dir}/{name}-desktop.png', full_page=True)
            print(f'✅ {name} (desktop)')
            await page.close()
        await desktop_context.close()
        
        # Mobile 390x844 (iPhone 12)
        mobile_context = await browser.new_context(viewport={'width': 390, 'height': 844})
        for url, name in pages:
            page = await mobile_context.new_page()
            await page.goto(url, wait_until='networkidle')
            await page.screenshot(path=f'{screenshots_dir}/{name}-mobile.png', full_page=True)
            print(f'✅ {name} (mobile)')
            await page.close()
        await mobile_context.close()
        
        await browser.close()
        print(f'\n✨ Screenshots saved to ./{screenshots_dir}/')

asyncio.run(capture_screenshots())

File: capture-nexus-screenshots.js
const playwright = require('playwright');
const fs = require('fs');
const path = require('path');

(async () => {
  const browser = await playwright.chromium.launch();
  const screenshotsDir = './screenshots';
  if (!fs.existsSync(screenshotsDir)) fs.mkdirSync(screenshotsDir);

  const pages = [
    { url: 'http://127.0.0.1:9696/nexus/', name: 'nexus-hub' },
    { url: 'http://127.0.0.1:9696/nexus/nexus-genesis.html', name: 'nexus-genesis' },
    { url: 'http://127.0.0.1:9696/nexus/nexus-autonomy.html', name: 'nexus-autonomy' },
    { url: 'http://127.0.0.1:9696/nexus/nexus-discovery.html', name: 'nexus-discovery' },
    { url: 'http://127.0.0.1:9696/nexus/nexus-integration.html', name: 'nexus-integration' }
  ];

  // Desktop
  const desktopCtx = await browser.createContext({ viewport: { width: 1920, height: 1080 } });
  for (const page of pages) {
    const p = await desktopCtx.newPage();
    await p.goto(page.url, { waitUntil: 'networkidle' });
    await p.screenshot({ path: `${screenshotsDir}/${page.name}-desktop.png`, fullPage: true });
    console.log(`✅ ${page.name} (desktop)`);
    await p.close();
  }
  await desktopCtx.close();

  // Mobile (iPhone 12)
  const mobileCtx = await browser.createContext({ viewport: { width: 390, height: 844 } });
  for (const page of pages) {
    const p = await mobileCtx.newPage();
    await p.goto(page.url, { waitUntil: 'networkidle' });
    await p.screenshot({ path: `${screenshotsDir}/${page.name}-mobile.png`, fullPage: true });
    console.log(`✅ ${page.name} (mobile)`);
    await p.close();
  }
  await mobileCtx.close();

  await browser.close();
  console.log(`\n✨ Screenshots saved to ./${screenshotsDir}/`);
})();
